tests/gtests/internals/sdpa_internal.hpp

```diff
@@ -0,0 +1,182 @@
+
+namespace dnnl {
+namespace impl {
+
+/// Scaled Dot Product Attention (sdpa) primitive.
+struct sdpa : public dnnl::primitive {
+    /// Primitive descriptor for a sdpa primitive.
+    struct primitive_desc : public dnnl::primitive_desc {
+        /// Default constructor. Produces an empty object.
+        primitive_desc() = default;
+
+        /// Constructs a primitive descriptor for a sdpa primitive
+        ///
+        /// @param aengine Engine to use.
+        /// @param query_desc Memory descriptor for query tensor.
+        /// @param key_desc Memory descriptor for key tensor.
+        /// @param value_desc Memory descriptor for value tensor.
+        /// @param output_desc Memory descriptor for output tensor.
+        /// @param attr Primitive attributes to use. Attributes are optional
+        ///     and default to empty attributes.
+        /// @param kq_attr Primitive attributes to use. Attributes are optional
+        ///     and default to empty attributes.
+        /// @param vs_attr Primitive attributes to use. Attributes are optional
+        ///     and default to empty attributes.
+        primitive_desc(const engine &aengine, const memory::desc &query_desc,
+                const memory::desc &key_desc, const memory::desc &value_desc,
+                const memory::desc &output_desc,
+                const primitive_attr &attr = default_attr(),
+                const primitive_attr &kq_attr = default_attr(),
+                const primitive_attr &vs_attr = default_attr())
+            : primitive_desc(aengine, query_desc, key_desc, value_desc, nullptr,
+                    memory::data_type::undef, output_desc, false, 1, false,
+                    attr, kq_attr, vs_attr) {}
+
+        /// Constructs a primitive descriptor for a sdpa primitive
+        ///
+        /// @param aengine Engine to use.
+        /// @param query_desc Memory descriptor for query tensor.
+        /// @param key_desc Memory descriptor for key tensor.
+        /// @param value_desc Memory descriptor for value tensor.
+        /// @param output_desc Memory descriptor for output tensor.
+        /// @param attn_mask_desc Memory descriptor for attention mask tensor.
+        /// @param attr Primitive attributes to use. Attributes are optional
+        ///     and default to empty attributes.
+        primitive_desc(const engine &aengine, const memory::desc &query_desc,
+                const memory::desc &key_desc, const memory::desc &value_desc,
+                const memory::desc &attn_mask_desc,
+                const memory::desc &output_desc,
+                const primitive_attr &attr = default_attr(),
+                const primitive_attr &kq_attr = default_attr(),
+                const primitive_attr &vs_attr = default_attr())
+            : primitive_desc(aengine, query_desc, key_desc, value_desc,
+                    &attn_mask_desc, memory::data_type::undef, output_desc,
+                    false, 1, false, attr, kq_attr, vs_attr) {}
```

Comment:
I would advocate towards a single `primitive_desc` interface (the one below) to avoid potential misuse and more code updates as long as it changes over time.